tools/space-usage/README.rst

```diff
@@ -0,0 +1,58 @@
+Space Reporting in JSON Format
+==============================
+
+If a site uses XRootD and WebDAV doors provided by native software (e.g. `XRootD <http://xrootd.org/>`_)
+without running a full suite of Grid middleware (such as `dCache <https://www.dcache.org/>`_), space reporting has to be provided externally.
+
+This is facilitated with a JSON file which the site has to update at least every 2 hours (e.g. via cron job).
+
+
+Generating the file
+-------------------
+Sample content of this file is provided within this folder of the repository in
+`space-usage-sample.json <https://github.com/rucio/rucio/blob/master/tools/space-usage/space-usage-sample.json>`_.
+
+The approach to retrieve the space-usage data is necessarily site-specific, if the underlying file system is a POSIX parallel cluster filesystem (e.g. BeeGFS, Lustre), the best performing way is usually to rely on the quota accounting provided by the file system.
+
+It should be noted that the ``num_files`` field is optional, which may help in the situation that no central accounting is available and the full file tree has to be walked to extract the space usage information.
+
+All units in the `space-usage-sample.json <https://github.com/rucio/rucio/blob/master/tools/space-usage/space-usage-sample.json>`_ file are in *bytes*. A Python script and schema for easy validation are also provided.
+
+
+Providing the File
+------------------
+The space usage file has to be provided via HTTPS, usually this can be easily done using an existing WebDAVs door.
+For example, the file can be exported from the exported endpoint directory.
+
+Concerning permissions, the following restrictions can be made:
+ - Readable by ``VO ATLAS`` with ``Role=production``.
+ - Only accessible from host ``rucio-nagios-prod-02``.
+
+In case WebDAVs is used, you can test using the following commands on a site with `ATLAS Local Root Base <https://twiki.atlas-canada.ca/bin/view/AtlasCanada/ATLASLocalRootBase>`_ available (for example, via CVMFS, or use *lxplus*) in case you have the appropriate permissions with your VOMS proxy::
+
+  setupATLAS
+  lsetup rucio
+  voms-proxy-init -voms atlas
+  lsetup davix
+  davix-get -P Grid https://my-example-size.com:8443/atlas/atlaslocalgroupdisk/space-usage.json
+
+
+AGIS Configuration
+------------------
+If the space reporting is set up, it has to be registered in `AGIS <https://atlas-agis.cern.ch/agis/>`_.
+
+For this, the **DDM Enpoint** has to be configured with the following settings:
+ - **Space method:** should be set to **other** (instead of **lcg-stmd**).
+ - **Space Usage:** should be set to the full HTTPS URL under which the ``space-usage.json`` file can be accessed.
+   This must include the necessary port (important e.g. in case WebDAVs is used).
+
+An example for an URL could be:
+`https://my-example-site.com:8443/atlas/atlaslocalgroupdisk/space-usage.json`
+
+Testing it works
+----------------
+To test the new space reporting is picked up, you can use the command::
```

Comment:
at the same place in the repo, there is the json schema and the a script to validate the syntax of the space usage file. We can add the information there:
validate-space-usage-json-file 
I also notice that the file https://github.com/rucio/rucio/blob/master/tools/space-usage/space-usage-schema.json can be extended with a description for each field